### PR TITLE
 Add filterable flag to ChangeLogParameter class (DAT-18811)

### PR DESCRIPTION
**Pull request by abrackx**
## Impact

- [x] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 

## Description
Add new `systemParameters` field to `ChangeLogParameters`. Respect new `filterable` flag when filtering changelog parameters. System parameters such as environment variables do not get filtered as there is no way to configure a label or context for these types of parameters. This enables users to use label filters such as `@some-filter` and still have values replaced from global changelog parameters like environment variables. 

## Things to be aware of
This pr treats every "system" parameter like environment variables as "not filterable". This is respected ONLY when these "system" parameters are set up without any filterable attributes (dbms, label, context).

### CODE REVIEW

File: liquibase-standard/src/test/groovy/liquibase/changelog/ChangeLogParametersTest.groovy [hunk 1]
```diff
@@ -145,4 +145,17 @@ class ChangeLogParametersTest extends Specification {
         then:
         expanded == "1234"
     }
+
+    def "System properties set up naively are not filtered when using required labels or contexts"() {
+        when:
+        def database = new MockDatabase()
+
+        System.setProperty("SOME_PROPERTY", "This should not be filtered")
```
Comment (StevenMassaro) [inline liquibase-standard/src/test/groovy/liquibase/changelog/ChangeLogParametersTest.groovy:153]: You should unset this System property in a cleanup block in this test.

File: liquibase-standard/src/main/java/liquibase/changelog/ChangeLogParameters.java
Comment (StevenMassaro) [inline liquibase-standard/src/main/java/liquibase/changelog/ChangeLogParameters.java:1]: Could you add some javadoc around the changes you made to this class? It's already a bit confusing.

**Review by StevenMassaro — CHANGES_REQUESTED** [review]

File: liquibase-standard/src/main/java/liquibase/changelog/ChangeLogParameters.java
Comment (abrackx) [reply liquibase-standard/src/main/java/liquibase/changelog/ChangeLogParameters.java:1]: Done!

File: liquibase-standard/src/test/groovy/liquibase/changelog/ChangeLogParametersTest.groovy (hunk 1 above)
Comment (abrackx) [reply liquibase-standard/src/test/groovy/liquibase/changelog/ChangeLogParametersTest.groovy:153]: Done!

File: liquibase-standard/src/main/java/liquibase/changelog/ChangeLogParameters.java
Comment (abrackx) [reply liquibase-standard/src/main/java/liquibase/changelog/ChangeLogParameters.java:1]: Updated javadocs and added comments!

**Review by filipelautert — APPROVED** [review]

**Pull request merged**